### PR TITLE
[DO NOT MERGE] Add tests related to SCM feature (scm_folder and SVN monorepo)

### LINKED ISSUE #4218: Optimization on SCM ('scm_folder.txt' pointing to local folder) could lead to unexpected package
When the SCM feature is used, there is an optimization to avoid the git commands `git commit`,  `git push` in order to test new sources for `conan create` command: Conan creates a file 'scm_folder.txt' inside the cache that stores the path to the local sources and copies them from it instead of cloning the repository.

This optimization also applies for `conan install <...> --build=ref/v1@user/channel`, so this command will copy the sources from the local folder in order to build `ref` (but those sources could correspond to a different version, package,... to the one supposed to be at the moment when the recipe was created in the cache).

### LINKED ISSUE #4222: [SVN] Optimization on SCM ('scm_folder.txt') breaks url "auto" on monorepo (some use-cases)
SVN allows the user to checkout a single directory from a repository instead of the full one from the repo root. Also, it is common practice in SVN to have several projects inside the same repository. Given these two circumstances, there is a function in the SVN class that may return different values for the _same_ repository (as it can be a different _working copy_):
 * SVN::get_repo_root: local folder where the root of the repo is located. Depending on the checked out url, the _conanfile.py_ will be in a different relative path.
 * SVN::get_remote_url: url corresponding to the place where the SVN is invocated (by default it will be the _connafile.py_ directory)

This is the reason why SVN has some warnings/notes in the docs about these functions [[1]](https://docs.conan.io/en/latest/reference/tools.html#tools-svn) [[2]](https://docs.conan.io/en/latest/creating_packages/package_repo.html)

These functions are used in the context of the `scm_folder.txt` optimization:

```python
    scm = SCM(scm_data, conanfile_dir, output)  # <-- Instantiated in 'conanfile_dir'
    if get_bool_from_text(os.environ.get("USE_SCM_FOLDER", "True")):
        if scm_data.capture_origin or scm_data.capture_revision:
            src_path = scm.get_repo_root()  #!!! Depends on the URL used to check out
            save(scm_src_file, src_path.replace("\\", "/"))

    if scm_data.url == "auto":
        origin = scm.get_qualified_remote_url()  #  <-- Calls get_remote_url()
        if not origin:
            raise ConanException("Repo origin cannot be deduced by 'auto'")
        if scm.is_local_repository():
            output.warn("Repo origin looks like a local path: %s" % origin)
        output.success("Repo origin deduced by 'auto': %s" % origin)
        scm_data.url = origin
```

The value of `scm["url"]` for SVN (it uses `get_remote_url()`) should be assigned explicitly (not using `"auto"`). It is on the user side, as there is no way to know it automatically. The user should implement his/her own function, as proposed in documentation here https://docs.conan.io/en/latest/creating_packages/package_repo.html. If the `"auto"` is used, then it will be resolved to the URL corresponding to the _conanfile.py_ directory (SCM is instantiated there: `scm = SCM(scm_data, conanfile_dir, output)`)... and it won't be the desired path in the general use case.

So, in the tests proposed in #4193, the contents of the source folder will be different if local path in `scm_folder.txt` is used (it will contain the path to the root of the repo), or if the `scm["url"]` is used (it will point to the place where the _conanfile.py_ file is).

### PR DESCRIPTION
**Pull request by jgsogo**
Changelog: (Feature | Fix | Bugfix): Describe here your pull request
Docs: https://github.com/conan-io/docs/pull/XXXX

@conan-io/barbarians, here are two tests that are failing related to SCM feature (the optimization of the `scm_folder.txt` file). I want you to take a look at them:

 * **SCMFolderObsoleteTest**: scm_folder.txt will copy sources from local workspace, there can be unexpected files (a different version of the same library, in this test) 🤕 
* **SCMFolderSVNCheckout**: several tests here, using the same conanfile in different contexts, local workflow and `conan create ..` one, from the same folder and from a different one,... some use-cases related to the _monorepo_ checkout fails, others print a `is not a working copy` SVN warning.
* **SCMFolderGitTest**: creation in cache with nested dirs fails, but it may be ok.

Have a look to: #3831 (SVN warning, last comment),  #3786 

If fixed, it should close #4218 and close #4222

### CODE REVIEW

File: conans/client/cmd/export.py [hunk 1]
```diff
@@ -76,10 +77,11 @@ def _capture_export_scm_data(conanfile, conanfile_dir, destination_folder, outpu
     captured_revision = scm_data.capture_revision
 
     scm = SCM(scm_data, conanfile_dir, output)
-    if scm_data.capture_origin or scm_data.capture_revision:
-        # Generate the scm_folder.txt file pointing to the src_path
-        src_path = scm.get_repo_root()
-        save(scm_src_file, src_path.replace("\\", "/"))
+    if get_bool_from_text(os.environ.get("USE_SCM_FOLER", "True")):
```
Comment (jgsogo) [inline conans/client/cmd/export.py]: These changes are here just to be able to run tests without the `scm_folder.txt` file.

Comment (memsharded) [reply conans/client/cmd/export.py]: USE_SCM_FOLER => USE_SCM_FOLDER?

**Comment by memsharded** [comment]
Agree this is important, and need some thinking and fixes.

File: conans/client/cmd/export.py (hunk 1 above)
Comment (lasote) [reply conans/client/cmd/export.py]: If this is only for testing I don't like it, but we could consider this a feature naming in properly: `CONAN_USE_SCM_FOLDER`. To discuss.

Comment (jgsogo) [reply conans/client/cmd/export.py]: Right now it is only to make these tests possible. I want you all to review the tests, check the failing ones, they are showing some bugs that cannot be fixed without adding something else to the model.

**Comment by jgsogo** [comment]
These tests, fixes and implementation are being done in:

#4249 

#4301

**Pull request closed without merging**